File: Sorting/Bubble Sort/java/BubbleSort.java

public class BubbleSort {

  static void sort(int[] array) {
    for (int i = 0; i < array.length; i++) {
      for (int j = 0; j < array.length - 1; j++) {
        if (array[j] > array[j + 1]) {
          int temp = array[j];
          array[j] = array[j + 1];
          array[j + 1] = temp;
        }
      }
    }
  }

  public static void main(String[] args) {
    int[] array = { 5, 3, 8, 6, 2 };
    sort(array);
    System.out.println("Sorted array");
    for (int num : array) {
      System.out.print(num + " ");
    }
  }
}

File: Sorting/Bubble Sort/cpp/bubble_sort.cpp
#include <iostream>
using namespace std;

void sort(int array[], int n) {
    for (int i = 0; i < n - 1; i++) {
        for (int j = 0; j < n - i - 1; j++) {
            if (array[j] > array[j + 1]) {
                swap(array[j], array[j + 1]);
            }
        }
    }
}

void printArray(int array[], int n) {
    for (int i = 0; i < n; i++) {
        cout << array[i] << " ";
    }
    cout << endl;
}

int main() {
    int array[] = {5, 3, 8, 6, 2};    
    int n = sizeof(array) / sizeof(array[0]);
    
    sort(array, n);

    cout << "Sorted array: ";
    printArray(array, n);
    
    return 0;
}
